src/renderer/library/components/opds/SearchForm.tsx

```diff
@@ -95,11 +98,23 @@ class SearchForm extends React.Component<IProps, undefined> {
                     className="R2_CSS_CLASS__FORCE_NO_FOCUS_OUTLINE"
                 />
                 {/* <button
-                    disabled={!this.props.search?.url}
-                    title={__("header.searchTitle")}
+                disabled={!this.props.search?.url}
+                title={__("header.searchTitle")}
+            >
+                <SVG ariaHidden={true} svg={SearchIcon} />
+            </button> */}
+                {this.props.search?.url ?
+                <button
+                    className={stylesGlobal.button_clear}
+                    type="reset"
+                    onClick={() => {
+                        this.inputRef.current.value = null;
```

Comment:
Code safety and happy compiler: `this.inputRef.current?.value =`
...and I think in this case writing directly into the DOM is okay as there is no change event handler or React state management attached to the input field.

Reply:
TypeScript does not allow this: "La partie gauche d'une expression d'assignation ne doit pas être un accès à une propriété facultative.ts(2779)"